Cura 7222 get coordinates in camera tool

jellespijker opened the pull request:
Camera can now be manipulated from a dialog either by hand or bij moving the mouse around.

jellespijker committed:
Updated qml import version to use Qt5.10

CURA-7222

jellespijker committed:
Removed calls to controller

CURA-7222

No need anymore to use the CustomCameraView, and the controller in
Uranium. Major rewrite

jellespijker committed:
Simplified Camera tool interface

It now shows only 1 view, storing and loading not further pursued at
this time. Currently x,y,z,pitch, roll and yaw implemented.

CURA-7222

jellespijker committed:
Added projection and zoom to the controls

Extra controls for the projection and zoom are added to the window.
note: The zoom factor seems to be initialized with a negative number
somewhere??

CURA-7222

jellespijker committed:
Perspective mode change stored also in preference

CURA-7222

jellespijker committed:
Checkbox now reflects the actual state

CURA-7222

jellespijker committed:
init of member variable w typing Optional[] = None is python 3.5 comp.

Made the initialization for _manager, _controller compatible with
Python 3.5 typeing, also the reference in the _createPlugin

CURA-7222

jellespijker committed:
init of member variable w typing Optional[] = None is python 3.5 comp.

Made the initialization for _manager, _controller and _camera with
compatible with Python 3.5 typeing

CURA-7222

jellespijker committed:
Merge remote-tracking branch 'origin/CURA-7222_get_coordinates_in_camera_tool' into CURA-7222_get_coordinates_in_camera_tool

# Conflicts:
#	src/CameraPosition.py

jellespijker committed:
Fixed recursive property setting

CURA-7222

jellespijker committed:
Validator zoom field is caped at a sane value

CURA-7222

jellespijker committed:
Cleanup of unused name attribute

CURA-7222

jellespijker committed:
Bug fix setting the perspective

It called a setPerspective function which didn't exist
CURA-7222

jellespijker committed:
Clean up unused code

Allowing to set all camera properties at once. This is no longer used
but could be usefull for restoring a safe view.

CURA-7222

jellespijker committed:
Window should now stay on top

Added WindowStaysOnTopHint flag this should enable the toolwindow to
stay on top

CURA-7222

ninovanhooff commented on resources/qml/CameraPositionPanel.qml:
```diff
@@ -104,7 +104,7 @@ UM.Dialog
             text: "zoom"
             value: manager.zoom
-            validator: DoubleValidator {bottom: -10; top: 10;}
+            validator: DoubleValidator {bottom: -0.495; top: 10;}
```
unnamed but especially "Weird" hard-coded values benefit from a comment line

jellespijker replied:
The value is obtained because it is hard coded in the the Uranium CameraTool.py line [318](https://github.com/Ultimaker/Uranium/blob/c68020a0f8eb974e7ccd67c8430e9ed1acec981a/plugins/Tools/CameraTool/CameraTool.py#L318). 

Which is presumably set to guard the limit when for the function _zoomCamera when zoomfactor goes to -0.5. At that point a division bij zero occurs and the function domain below  -0.5 is undefined.

jellespijker committed:
Clarified hardcoded lower limit in for the Validator

[See commet](https://github.com/Ultimaker/cura-camera-position/pull/1#discussion_r438750124)
CURA-7222
